client/src/components/Users/Auth/SignUpCitizen/StepOne.tsx

```diff
@@ -0,0 +1,296 @@
+import React, { useState, useEffect } from 'react';
+import { Box, Button, Checkbox, Typography, FormControlLabel } from '@mui/material';
+import { string, boolean } from 'yup';
+import { useQuery } from 'react-query';
+
+import StyledLink from '../../../StyledLink';
+import routes from '../../../../routes/routes';
+import PasswordInput from '../PasswordInput';
+import EmailInput from '../EmailInput';
+import NameInput from '../NameInput';
+import { ValidationUtils } from '../../../../utils';
+import Endpoints from './apis/backend';
+
+import { useStyles } from './styles/styles';
+
+const initialFormData = {
+  firstName: {
+    value: '',
+    error: null,
+    rule: string().required('Required.'),
+  },
+  lastName: {
+    value: '',
+    error: null,
+    rule: string().required('Required.'),
+  },
+  email: {
+    value: '',
+    error: null,
+    rule: string().email('Invalid email.').required('Required.'),
+  },
+  password: {
+    value: '',
+    error: null,
+    rule: string()
+      .min(8, 'Password is too short - should be 8 chars minimum.')
+      .required('Required.'),
+  },
+  passwordConfirm: {
+    value: '',
+    error: null,
+    rule: string()
+      .min(8, 'Password is too short - should be 8 chars minimum.')
+      .required('Required.'),
+  },
+  acceptTerms: {
+    value: false,
+    error: null,
+    rule: boolean().required('The terms and conditions must be accepted.'),
+  },
+};
+
+interface StepOneType {
+  initData: {};
+  handleNext: (formData: {}) => void;
+}
+
+export default function StepOne({ initData, handleNext }: StepOneType) {
+  const { classes } = useStyles();
+
+  Object.keys(initialFormData).forEach((key) => {
+    //@ts-ignore
+    initialFormData[key].value = initData[key];
+  });
+
+  const [formData, setFormData] = useState(initialFormData);
+  const [userEmailQueryEnabled, setUserEmailQueryEnabled] = useState(false);
+
+  const { status, data } = useQuery({
+    queryKey: ['user-email-exists', formData['email'].value],
+    queryFn: () => Endpoints.checkUserEmail(formData['email'].value),
+    enabled: userEmailQueryEnabled,
+  });
```

Comment:
I think we should leave this query out of this iteration until we define this further. What do you think?